### ISSUE TITLE
DF energy of 2D case of example pbc/31-low_dimensional_pbc.py

### ISSUE THREAD

**Issue by maxnus** [pyscf/pyscf#558]
If I run the example `pbc/31-low_dimensional_pbc.py`, the final output for the 2D case is

> 2D:  FFT      AFT      DF       MDF       super-mole
[-1.0504972460759572, -1.066843008736809, -0.1459582968108064, -1.0689448036230722, -1.060859818915696]

The third number - for DF - seems off compared to the others.
Is this expected?

Full output attached below.

[output.txt](https://github.com/pyscf/pyscf/files/4344248/output.txt)

**Comment by sunqm** [comment]
The size of vacuum is too small in this example. You can get reasonable value
for a bit larger vacuum.
```
a = [[L,0,0],[0,L,0],[0,0,15]],
```

Truncated Coulomb interaction is now the default choice for 2D system. It has
certain requirements on the size of vacuum. This example was created with a
different Coulomb treatment which has no spacial requirement on vacuum. It is
out of date we should update the example soon.

**Comment by Majokux** [comment]
> The size of vacuum is too small in this example. You can get reasonable value for a bit larger vacuum. ``` a = [[L,0,0],[0,L,0],[0,0,15]], ``` Truncated Coulomb interaction is now the default choice for 2D system. It has certain requirements on the size of vacuum. This example was created with a different Coulomb treatment which has no spacial requirement on vacuum. It is out of date we should update the example soon.

Thanks for the quick reply. What's the downside of having a large vacuum, i.e. why is it not set automatically to a value of 1000 or similar for 2D systems?

**Comment by gkc1000** [comment]
If you use FFTDF, the number of plane waves is then enormous along the
vacuum direction making the calculation slow.

**Comment by Majokux** [comment]
OK mthanks. Is the way `cell.mesh` is calculated automatically from `cell.precision` a good default even for 2D systems? I get almost 600 points in `mesh[2]` for the recommended vacuum size for my system (around 30 A).

**Comment by sunqm** [comment]
Yes, cell.mesh is estimated based on the basis set, the size of the unit cell and the required accuracy. The estimated vacuum size is quite conservative. Often reducing ~30% of that value does not lead changes in accuracy.

For that particular example, the estimated value is ~30 Bohr or 15 Angstrom. I ran some tests. The vacuum of 10 A still works for GDF.

In GDF, we introduced a smooth functions to screen Coulomb integrals. This worsen the convergence of vacuum size, compared to other DF methods. We will in the future improve the GDF algorithm to reduce this negative effects.
